### ISSUE TITLE
Add spawn note to RevivalUnderGodTime actors in area 64

### ISSUE THREAD

**Issue by leoetlino** [zeldamods/objmap#1]
Some map unit actors in area 64 only spawn when it's "god time" i.e. when the Lord of the Mountain appears (AnimalMaster_Appearance is true).

This should be noted on the object map to avoid confusion.

* [x] Write a beco parser (for FieldMapArea)
* [x] Make a list of actor tags
* [ ] Have ObjectInfo check the object's associated map area; if it is Field Map Area 64, check whether the actor has the RevivalUnderGodTime tag (TODO: verify) and show a note if they do have it.

**Comment by savage13** [comment]
So beco parser is done.

I am unsure about the list of Actor Tags.  `RevivalUnderGodTime` is only appears to be associated with a `Animal_RupeeRabbit_A`.  There are `RevivialRandom` and a `UnderGodForest` tags
 (and a `UnderGodForestOff` tag).

Along the same lines, os it possible to determine if an object / animal is "Night only"? `Animal_Night` and `TeamStalfos` are possible.

**Comment by leoetlino** [comment]
Oh yeah, then the relevant actor tag is probably UnderGodForest, not RevivalUnderGodTime (though the latter is related to the LotM). Not sure why I wrote RevivalUnderGodTime 3 years ago...

> Along the same lines, os it possible to determine if an object / animal is "Night only"? Animal_Night and TeamStalfos are possible.

Unfortunately that's more difficult, because time-based spawn restrictions are typically implemented with map object links.

Here's an example: https://objmap.zeldamods.org/#/map/z7,-438,1113?id=MainField,E-6,3736753824

**Comment by savage13** [comment]
Was able to get the fieldArea into the map database from the Beco files.

The Tags  need to be parsed out of the ActorLink files, but that could be done based on the `UnitConfigName` and a HashMap.  Is it possible that other Tags will be wanted?

Would it be best to insert it into the `!Parameters` section of each object?

**Comment by leoetlino** [comment]
> Is it possible that other Tags will be wanted?

Perhaps

> Would it be best to insert it into the !Parameters section of each object?

IMO we should avoid modifying the original map data. Couldn't we put the tags in a separate table column instead?

**Comment by savage13** [comment]
Parsed *all* the tags into a separate table that looks like `create table ( actor_name text, tags json )`  or possibly `create table ( actor_name text, tag text )` 

To get the data, a number of options are:
- join on the tags table for each request and send all tags (this is easy)
- send a boolean of `(field_area = 64 and  'UnderGodForest' in tags)` (this is more SQL)
- something else ?

This depends on where the logic / processing should be and how much data should be sent.

I kinds like the 2nd option as it only produces a single value like `LordOfTheMountain` in the JSON.  The SQL is not  complex written as a subquery on the tags table (2nd version). The down side is that it is not extensible, but that might be a question for later.

```
  (select (count(tag) > 0) from tags where actor_name = unit_config_name and tag = 'UnderGodForest') 
       * (field_area = 64) as lotm,
```

**Comment by leoetlino** [comment]
Yeah, I think the second option is fine. When we send object data we would just have to add one new property ("spawns_with_lotm"?)

**Comment by savage13** [comment]
Ok. Sounds good

Dumb question: How sure are we that all of the objects are linked to this Tag?  There was a question about if specific Acorns show up during LotM. Acorns do not have the UnderGodForest tag.

Thoughts on the Horsey?

![Screenshot 2021-09-09 at 19-44-22 BotW Object Map](https://user-images.githubusercontent.com/126514/132776664-73c22a1d-e42f-4b08-be1c-e3711515ab95.png)

**Comment by leoetlino** [comment]
The spawn check is based on the UnderGodForest and UnderGodForestOff tags (see 0x7100d4d760 in Switch 1.5.0). Are we sure those acorns really only spawn when the LotM shows up?

The icon looks fine -- the bluish color is a nice touch!

**Comment by leoetlino** [comment]
fixed by #40
